packages/core/hash.ts

```diff
@@ -10,7 +10,8 @@ export function hashValues(values: Buffer[]): Buffer {
 }
 
 export function hashValuesWithSalt(values: Buffer[]): { hash: Buffer; salt: Buffer } {
-  const salt = randomBytes(16);
+  // Salt size should be the same as the hash size (32 bytes for SHA3_256)
+  const salt = randomBytes(SHA3_256Hasher.size / 8);
```

Comment:
tldr : we can simply define  `const salt = randomBytes(32);` 
Or with more details `const salt = randomBytes(2 * securityLevel);` with `securityLevel = 128` 

There isn't a direct correlation between the salt size and the hash size in this case. 
The hash size is about the output space, while the salt size is about the input space. 
Here we would like the input space to be "large enough" so it's not "practical" for:
 - a user to compute the hash of all inputs (bounded by the size of the set)
 - a minter to find two inputs with the same hash (bounded by the square root of the set size)

"large enough" is commonly defined as 2^128, and that gives an input set of 2^256 or just 256 bits.

Reply:
Thanks, that makes sense!

I simplified it to `randomBytes(32)`